joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/datastructures/ProgramSummary.scala

```diff
@@ -43,6 +43,7 @@ object ProgramSummary {
   /** Combines two namespace-to-type maps.
     */
   def combine[T <: TypeLike[_, _]](a: Map[String, Set[T]], b: Map[String, Set[T]]): Map[String, Set[T]] = {
+    //fixme: Use mutable datatypes, otherwise folds are quadratc
```

Comment:
Is this more suggesting that we'd want something more like `DiffGraphBuilder.absorb`?